lib/addons/src/types.ts

```diff
@@ -253,10 +253,20 @@ export interface BaseMeta<ComponentType> {
 }
 
 export interface BaseStory<Args, StoryFnReturnType> {
-  (args: Args, context: StoryContext): StoryFnReturnType;
-
+  (args: Args | StoryContext, context?: StoryContext): StoryFnReturnType;
```

Comment:
to work for both legacy and new story formats which are used in the composeStory function:
`story(args, context) || story(context)`